sensor/kubernetes/crs/crs.go

```diff
@@ -0,0 +1,300 @@
+package crs
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	metautils "github.com/grpc-ecosystem/go-grpc-middleware/v2/metadata"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/booleanpolicy/policyversion"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/crs"
+	"github.com/stackrox/rox/pkg/env"
+	grpcUtil "github.com/stackrox/rox/pkg/grpc/util"
+	"github.com/stackrox/rox/pkg/k8sutil"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/mtls"
+	"github.com/stackrox/rox/pkg/pods"
+	protoconv "github.com/stackrox/rox/pkg/protoconv/certs"
+	"github.com/stackrox/rox/pkg/protoutils"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/version"
+	"github.com/stackrox/rox/sensor/common/centralclient"
+	sensorCommon "github.com/stackrox/rox/sensor/common/sensor"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh"
+	"github.com/stackrox/rox/sensor/kubernetes/certrefresh/securedcluster"
+	"github.com/stackrox/rox/sensor/kubernetes/helm"
+	"github.com/stackrox/rox/sensor/kubernetes/sensor"
+	"google.golang.org/grpc/metadata"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	log                                  = logging.LoggerForModule()
+	fetchSensorDeploymentOwnerRefBackoff = wait.Backoff{
+		Duration: 10 * time.Millisecond,
+		Factor:   3,
+		Jitter:   0.1,
+		Steps:    10,
+		Cap:      6 * time.Minute,
+	}
+)
+
+// EnsureServiceCertificatesPresent initiates the CRS-based cluster registration flow to retrieve the
+// service certificates in case no services certificates are found and instead a CRS is present.
+func EnsureServiceCertificatesPresent() error {
+	log.Infof("Ensuring certificates for Secured Cluster services are present.")
+	clientconn.SetUserAgent(fmt.Sprintf("%s Cluster Registration Helper", clientconn.Sensor))
+
+	// Check if legacy sensor service certificate, e.g. created via init-bundle, exists.
+	if legacySensorServiceCert := os.Getenv(crs.LegacySensorServiceCertEnvName); legacySensorServiceCert != "" {
```

Comment:
I might be missing something here, but where do you check that the secret actually exists? (this seems to only check that an env var is not empty)

Reply:
Yes, this will be part of the follow-up PR, which does injects these certs as environment variables:

```
        - name: ROX_LEGACY_SENSOR_SERVICE_CERT
          # Keep in sync with pkg/crs/env.go.
          valueFrom:
            secretKeyRef:
              name: sensor-tls
              key: sensor-cert.pem
              optional: true
        - name: ROX_SENSOR_SERVICE_CERT
          # Keep in sync with pkg/crs/env.go.
          valueFrom:
            secretKeyRef:
              name: tls-cert-sensor
              key: cert.pem
              optional: true
```

This will allow the `EnsureServiceCertificatesPresent()` to check for existence of any of these certificates with a simple `os.Getenv()` check.

Reply:
Ah, got it. But why not check that the secrets exist in the code, using the k8s API?

Reply:
This way it keeps the `crs.go` mostly agnostic about naming of K8s resources.

Reply:
Also, if Kubernetes can do something for us and it means less code for us, I think that's worth considering. :-)